app/assets/javascripts/common/password_validation.js

```diff
@@ -39,15 +30,41 @@ PasswordValidation = function() {
         "X-CSRF-Token": token
       },
       data: {"password": password}
-    }).done(function(data, status){
-      console.log(status)
-      console.log(data)
-      this.timer = null;
+    }).done((data, status) => {
+      let response;
       if (status === "success") {
-        this.response = data["errors"];
+        response = data["errors"];
       } else {
-        this.response = null;
+        // if we don't get a response, mark all validations as failures
```

Comment:
This means if the server returns an error because of some other failure, we will display it as a password error to the user?  It may be better to show a generic error to the error if that happens, because it would mean something is wrong on the server side (a 500 error for example).  Also, a stale session would result in this case, because the CSRF token would've expired...we could even just show some sort of generic error w/ the HTTP code so users could pass it along if they are trying to get help from support:

```
An error occurred while trying to validate your password (HTTP 500)
```

Reply:
I'll talk to Claudio about that since it will require some visual changes. Thankfully, I wrote a small ticket that's coming after this for the case when the reset link is expired, so we just need to add some new copy. 

As for the nature of the error... I think the CSRF expiring would be a very edge case since I think it would mean the person has been sitting on the password setting page for hours. But your point about handling the unknown possibilities better is a good one. TBH, I was working more on the assumption that the request above might fail due to connectivity issues, in which case the validations would just go back to "everything is invalid" and their next keystroke might go through and everything would work again. The downside to redirecting to an error page upon error here is that a single errant request effectively ends the attempt. I doubt this is a major concern.